src/actions/reports/cost.py

```diff
@@ -537,40 +714,27 @@ def collect_nodes_and_create_plots(self, specs: list[ChartSetSpec]):
                             spec.series_details[series_label] = set()
 
                         query, _ = self.query_map.get(query_str)
-                        spec.series_data[series_label].append(
-                            PlotSeriesData(xdata, cost, time_ms, node))
+                        spec.series_data[series_label].append(DataPoint(xdata, cost, time_ms, node))
 
                         cond = node.get_search_condition_str(with_label=True)
                         spec.series_details[series_label].add(f"{cond}" if cond
                                                               else "(No Search Condition)")
 
+        self.logger.info(f'Generating plots...')
 
-        colors = [ 'b', 'g', 'r', 'c', 'm', 'y', 'k' ]
-        marker_style = [ 'o', '8', 's', 'p', '*', '+', 'x', 'd',
-                    'v', '^', '<', '>', '1', '2', '3', '4',
-                    'P', 'h', 'H', 'X', 'D', '|', '_']
+        marker_style = [ '.', 'o', 'v', '^', '<',
+                         '>', '8', 's', 'p', '*',
+                         'h', 'H', 'D', 'd', 'P', 'X']
         line_style = [ '-', '--', '-.', ':' ]
 
         for spec in specs:
-            for i, (series_label, node_data) in enumerate(spec.series_data.items()):
-                node_data.sort(key=itemgetter(0,2,1)) # xdata,time,cost
-
-                fmt = colors[ i % len(colors) ]
-                fmt += marker_style[ i % len(marker_style) ]
-
-                if re.search('Seq Scan', series_label):
-                    fmt += ':'
-                elif re.search('Index Scan.*_pkey', series_label):
-                    fmt += '-'
-                elif re.search('Index Scan', series_label):
-                    fmt += '-.'
-                elif re.search('Index Only Scan', series_label):
-                    fmt += '--'
-                else:
-                    fmt += ':'
+            for i, (series_label, data_points) in enumerate(sorted(spec.series_data.items())):
+                fmt = self.get_series_color(series_label)
+                fmt += marker_style[ (i+3) % len(marker_style) ]
+                fmt += line_style[ (i+5) % len(line_style) ]
                 spec.series_format[series_label] = fmt
 
-            self.create_node_plots(spec)
+            self.draw_x_cost_time_plot(spec)
 
     def choose_chart_spec(self, chart_specs):
```

Comment:
This method should return `def choose_chart_spec(self, chart_specs) -> Mapping[ChartSpec]` according to this code:
```
        if interactive:
            chart_specs = report.choose_chart_spec(chart_specs)
        else:
....

        report.collect_nodes_and_create_plots(chart_specs)
        
        
         def collect_nodes_and_create_plots(self, specs: Mapping[ChartSpec]):
```

But we return list

Reply:
Oops, it's an error on the function signature side. It should be `specs: Iterable[ChartSpec]`.